tests/test_core.py

```diff
@@ -199,26 +196,110 @@ class test_TypeDict(unittest.TestCase):
 
     def test_init(self):
         """Test construction with different list types."""
-        raise NotImplementedError()
+        types = ('A','B')
+        default = {'A':1.0, 'B':1.0}
+
+        #test construction with tuple input
+        d = relentless.core.TypeDict(types=('A','B'))
+        self.assertEqual(d.types, types)
```

Comment:
Can you put back the checks here that the dict is filled with `None` if `default` is not specified? Also, let's not access `_data` directly. Use either `todict` or access by iterator. You could, for example, do this in a one-liner with list comprehension:
```py
self.assertEqual([d[t] for t in d.types], (None, None))
```